commands/service_execute.go

```diff
@@ -186,6 +173,15 @@ func (c *serviceExecuteCmd) readFile() (string, error) {
 	return string(content), err
 }
 
+func taskByKey(s *coreapi.Service, key string) *coreapi.Task {
```

Comment:
we can but we can check for the null as well, it's just a helper function only used inside this file.

Reply:
just seems better to adopt this pattern all the time. Also we are returning the error anyway just after so let's get the error from the function itself

Reply:
I feel it's not a really good pattern in Go to return a pointer without error.
I speak about this with Hubert about should we keep pointer in the data structure of service or not. We didn't find a good solution.
In both TypeScript and in Swift, they force to check if the variable is nil / undefined if it can be, before accessing it. This is not something that the Go compiler enforce. I feel it opens the place for many nil pointer error...

What about removing all pointers in the struct of the service? Like this, for example, this function `taskByKey`, will need to have the following signature: `taskByKey(s coreapi.Service, key string) (coreapi.Task, error)`.
Yeah it's will consume more memory, but we are forced to follow a good pattern.

Reply:
the pointer discussion is something else, the problem will be the same. Either we test something based on the value returned or based on an error.
After I kind of agree if it's pure data we don't necessary need pointers as long as it's not huge data and not really important that are present everywhere in all the modules. For this one I don't mind keeping the pointers. I don't see any really good reason to change it

Reply:
we can change to error, it's really small change (but error won't ever happen here) but it's ok